pkg/server/plugin/keymanager/v0_test.go

```diff
@@ -0,0 +1,359 @@
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo"},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): plugin response missing public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "response has malformed PKIX data",
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo", PkixData: []byte("malformed")},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): unable to parse public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "RPC fails",
+			err:           errors.New("ohno"),
+			expectCode:    codes.Unknown,
+			expectMessage: "keymanager(test): ohno",
+		},
+		{
+			test:       "success",
+			publicKey:  &keymanagerv0.PublicKey{Id: "foo", PkixData: testKeyPKIXData},
+			expectCode: codes.OK,
+		},
+	} {
+		tt := tt
+		t.Run(tt.test, func(t *testing.T) {
+			plugin := fakeV0Plugin{
+				getPublicKeyResponse: &keymanagerv0.GetPublicKeyResponse{
+					PublicKey: tt.publicKey,
+				},
+				getPublicKeyErr: tt.err,
+			}
+			km := loadV0Plugin(t, plugin)
+			key, err := km.GetKey(context.Background(), "foo")
+			spiretest.RequireGRPCStatusContains(t, err, tt.expectCode, tt.expectMessage)
+			if tt.expectCode != codes.OK {
+				return
+			}
+			require.NotNil(t, key)
+			assert.Equal(t, "foo", key.ID())
+			assert.Equal(t, testKey.Public(), key.Public())
+		})
+	}
+}
+
+func TestV0GetKeys(t *testing.T) {
+	for _, tt := range []struct {
+		test          string
+		err           error
+		publicKey     *keymanagerv0.PublicKey
+		expectCode    codes.Code
+		expectMessage string
+	}{
+		{
+			test:          "response missing PKIX data",
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo"},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): plugin response missing public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "response has malformed PKIX data",
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo", PkixData: []byte("malformed")},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): unable to parse public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "RPC fails",
+			err:           errors.New("ohno"),
+			expectCode:    codes.Unknown,
+			expectMessage: "keymanager(test): ohno",
+		},
+		{
+			test:       "success with no keys",
+			expectCode: codes.OK,
+		},
+		{
+			test:       "success with keys",
+			publicKey:  &keymanagerv0.PublicKey{Id: "foo", PkixData: testKeyPKIXData},
+			expectCode: codes.OK,
+		},
+	} {
+		tt := tt
+		t.Run(tt.test, func(t *testing.T) {
+			resp := &keymanagerv0.GetPublicKeysResponse{}
+			if tt.publicKey != nil {
+				resp.PublicKeys = []*keymanagerv0.PublicKey{tt.publicKey}
+			}
+			plugin := fakeV0Plugin{
+				getPublicKeysResponse: resp,
+				getPublicKeysErr:      tt.err,
+			}
+			km := loadV0Plugin(t, plugin)
+			keys, err := km.GetKeys(context.Background())
+			spiretest.RequireGRPCStatusContains(t, err, tt.expectCode, tt.expectMessage)
```

Comment:
it is intentional to always verify expected code on error?
it will verify that error is a codes.Ok, and message is empty.

Reply:
Answered that on different comment. I'll fix up the helper to assert an empty message on OK.